packages/ERTP/src/issuer.js

```diff
@@ -0,0 +1,240 @@
+      });
+    }
+
+    const totalPursesB = purses.map(purseLedger.get).reduce(add, empty);
+    const totalPaymentsB = payments.map(paymentLedger.get).reduce(add, empty);
+    const total = amountMath.add(totalPursesB, totalPaymentsB);
+
+    const newPaymentsTotal = newPaymentBalances.reduce(add, empty);
+    const newPursesTotal = newPurseBalances.reduce(add, empty);
+    const newTotal = amountMath.add(newPaymentsTotal, newPursesTotal);
+
+    // Invariant check
+    assert(amountMath.isEqual(total, newTotal), 'rights were not conserved');
+
+    // commit point
+    payments.map(payment => paymentLedger.delete(payment));
+    purses.map((purse, i) => purseLedger.set(purse, newPurseBalances[i]));
+
+    const newPayments = newPaymentBalances.map(balance => {
+      const newPayment = makePayment();
+      paymentLedger.init(newPayment, balance);
+      return newPayment;
+    });
+
+    return harden(newPayments);
+  };
+
+  const issuer = harden({
+    getBrand: () => brand,
+    allegedName: () => allegedName,
+    getAmountMath: () => amountMath,
+    getMathHelpersName: () => mathHelpersName,
+    makeEmptyPurse: () => {
+      const purse = makePurse();
+      purseLedger.init(purse, amountMath.getEmpty());
+      return purse;
+    },
+    isLive: paymentLedger.has,
+    getBalance: paymentLedger.get,
+    burn: (paymentP, optAmount = undefined) => {
+      const payment = E.unwrap(paymentP);
+      const paymentBalance = paymentLedger.get(payment);
+      assertAmountEqual(paymentBalance, optAmount);
+      paymentLedger.delete(payment);
+      return paymentBalance;
+    },
+    claim: (paymentP, optAmount) => {
+      const srcPayment = E.unwrap(paymentP);
+      const srcPaymentBalance = paymentLedger.get(srcPayment);
+      assertAmountEqual(srcPaymentBalance, optAmount);
+      // Commit point.
+      const [payment] = reallocate(
+        harden({
+          payments: [srcPayment],
+          newPaymentBalances: [srcPaymentBalance],
+        }),
+      );
+      return payment;
+    },
+    combine: (fromPaymentsPArray, totalAmount) => {
+      const fromPaymentsArray = fromPaymentsPArray.map(E.unwrap);
+      const totalPaymentsB = fromPaymentsArray
+        .map(paymentLedger.get)
+        .reduce(add, empty);
+      assertAmountEqual(totalPaymentsB, totalAmount);
+      // Commit point.
+      const [payment] = reallocate(
+        harden({
+          payments: fromPaymentsArray,
+          newPaymentBalances: [totalPaymentsB],
+        }),
+      );
+      return payment;
+    },
+    // payment to two payments, A and B
+    split: (paymentP, paymentAmountA) => {
+      const srcPayment = E.unwrap(paymentP);
+      paymentAmountA = amountMath.coerce(paymentAmountA);
+      const srcPaymentBalance = paymentLedger.get(srcPayment);
+      const paymentAmountB = amountMath.subtract(
+        srcPaymentBalance,
+        paymentAmountA,
+      );
+      const amounts = harden([paymentAmountA, paymentAmountB]);
```

Comment:
Excuse me if I asked that this array be hardened. `amounts` is only used once, in the reallocate call, where you're already hardening the enclosing record. Thus, you should just put this array inline there without its own harden, as is consistent with all the other array arguments to `reallocate` throughout this file. And avoid declaring any `amounts` variable.

Reply:
Good point, not sure where this came from.